docs/DeveloperGuide.md

```diff
@@ -1,34 +1,670 @@
+
+##### Implementation
+
+Below illustrates the steps taken by each command to ensure their execution.
+
+###### editNotes
+
+1. `checkEmptyList` would be called to ensure the `NoteList` is not empty before proceeding on with the execution. 
+If an empty list is detected, the command would terminate with a prompt of empty list to the user.
+1. A condition of whether the user has pre-inputted the title is checked.
+    * If the user pre-inputs a title, the findTitle function would go through the list to find the title. If the input 
+    title does not match any of the titles in the list, the command would terminate with a message: 
+    “title is not found”. If a match is found, the system prints out the note and asks for which part to edit. 
+    In the case of multiple matches, the system would print all matching cases and ask the user to select amongst them. 
+    Following the confirmation, a “T” would mean editing the title while “D” would mean editing the description.
+    * If the user did not pre-input the title, the system would print all notes currently in the list and ask for 
+the user to select which note to edit among them. Following the confirmation, a “T” would mean editing the title 
+while “D” would mean editing the description.
+       
+1. The new edited note would then be passed on to storage for saving.
+
+Figure 4.5-1 illustrates the above steps via a sequence diagram.
+
+![Figure 4.5-1](images/DeveloperGuide/Figure 4.5-1.png)
+_Figure 4.5-1: Sequence diagram for editNotesCommand_
+
+###### deleteNotes
+
+1. checkEmptyList would be called to ensure the NoteList is not empty before proceeding on with the execution. 
+If an empty list is detected, the command would terminate with a prompt of empty list to the user.
+1. A condition of whether the user has pre-inputted the title is checked.
+    * If the user has included a title in the command, the `findTitle` function would go through the list to 
+find the title. If the input title does not match any of the titles in the list, the command would terminate with 
+a message: “title is not found”. If a match is found, the system prints out the note and asks for confirmation 
+for deletion. In the case of multiple matches, the system would print all matching cases and ask the user to select 
+amongst them. Following the confirmation, a “N” would terminate the command while a “Y” would proceed to remove 
+the note from the list.
+    * If the user did not pre-input the title, the system would print all notes currently in the list and ask 
+for the user to select which note to delete among them. Following the confirmation for deletion, a “N” would terminate 
+the command while a “Y” would proceed to remove the note from the list.
+
+1. The current note list would then be saved by the `Storage` class.
+
+Figure 4.5-2 illustrates the above steps via a Sequence Diagram.
+
+![Figure 4.5-2](images/DeveloperGuide/Figure 4.5-2.png)
```

Comment:
Maybe you can add the end of the activation line (cross)  outside of the alt box to make your diagram clearer? 

<img width="764" alt="Screenshot 2020-10-29 at 12 56 34 PM" src="https://user-images.githubusercontent.com/57402349/97527461-538f3a00-19e6-11eb-8f9d-e51a89d1fd4c.png">

Reply:
![image](https://user-images.githubusercontent.com/60292763/97657627-5309a880-1aa5-11eb-9e3d-d3d2cb4a0883.png)

If the object is deleted, shouldn't there be continuing dashed line following it?